File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ListView/Programming/ItemFormatingValidatingPage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.ListView.Programming
{
    public partial class ItemFormatingValidatingPage : UserControl
    {
        public ItemFormatingValidatingPage()
        {
            InitializeComponent();

            //Set DataSource as array of Colors.
            KnownColor[] knownColors = (KnownColor[])Enum.GetValues(typeof(KnownColor));
            Color[] colors = new Color[knownColors.Length];
            for (int i = 0; i < knownColors.Length; ++i)
            {
                colors[i] = Color.FromKnownColor(knownColors[i]);
            }
            this.mobjForeCB.DataSource = colors;
            this.mobjForeCB.ColorMember = "Name";
            this.mobjForeCB.DisplayMember = "Name";
            
            this.mobjBackCB.DataSource = colors.Clone();
            this.mobjBackCB.ColorMember = "Name";
            this.mobjBackCB.DisplayMember = "Name";
            
            //Show Grid Lines
            this.mobjListView.GridLines = true;

            //Fill up the ListView
            this.mobjListView.Items.Add(new ListViewItem(new string[] { "User1", "8-800-1234556" }));
            this.mobjListView.Items.Add(new ListViewItem(new string[] { "User2", "8-800-9513546" }));
            this.mobjListView.Items.Add(new ListViewItem(new string[] { "User3", "8-800-8524563" }));
            this.mobjListView.Items.Add(new ListViewItem(new string[] { "User4", "8-800-9874613" }));
            this.mobjListView.Items.Add(new ListViewItem(new string[] { "User5", "8-800-1234556" }));
            this.mobjListView.Items.Add(new ListViewItem(new string[] { "User6", "8-800-9513546" }));
            this.mobjListView.Items.Add(new ListViewItem(new string[] { "User7", "8-800-8524563" }));
            this.mobjListView.Items.Add(new ListViewItem(new string[] { "User8", "8-800-9874613" }));
            this.mobjListView.Items.Add(new ListViewItem(new string[] { "User9", "8-800-1234556" }));
            this.mobjListView.Items.Add(new ListViewItem(new string[] { "User10", "8-800-9513546" }));
           
            //Set default background and foreground colors
            foreach (ListViewItem objItem in this.mobjListView.Items)
            {
                objItem.ForeColor = Color.Black;
                objItem.BackColor = Color.White;
            }
        }


        /// <summary>
        /// Handles the ItemFormatting event of the mobjListView control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="ListViewItemFormattingEventArgs"/> instance containing the event data.</param>
        private void mobjListView_ItemFormatting(object sender, ListViewItemFormattingEventArgs e)
        {
            //Define currently formatted item in ListView
            ListViewItem formattedItem = this.mobjListView.SelectedItem;
            if (formattedItem !=null)
            {
                //Change Result Text
                this.mobjResultLabel.Text = string.Format("Item #{0} has foreground {1} and background {2} colors.", formattedItem.Index, formattedItem.ForeColor.Name, formattedItem.BackColor.Name);
            }
            
        }

        /// <summary>
        /// Handles the SelectedIndexChanged event of the mobjForeCB control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjForeCB_SelectedIndexChanged(object sender, EventArgs e)
        {
            //Change fore color of selected item
            if (this.mobjListView.SelectedItem != null)
            {
                this.mobjListView.SelectedItem.ForeColor = (Color)this.mobjForeCB.SelectedItem;
            }
        }

        /// <summary>
        /// Handles the SelectedIndexChanged event of the mobjBackCB control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjBackCB_SelectedIndexChanged(object sender, EventArgs e)
        {
            //Change back color of selected item
            if (this.mobjListView.SelectedItem != null)
            {
                this.mobjListView.SelectedItem.BackColor = (Color)this.mobjBackCB.SelectedItem;
            }
        }


        /// <summary>
        /// Handles the SelectedIndexChanged event of the mobjListView control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjListView_SelectedIndexChanged(object sender, EventArgs e)
        {
            //Change selected items in ComboBoxes
            if (this.mobjListView.SelectedItem != null && this.mobjForeCB.Items.Contains(this.mobjListView.SelectedItem.ForeColor))
            {
                this.mobjForeCB.SelectedItem = this.mobjListView.SelectedItem.ForeColor;
            }
            if (this.mobjListView.SelectedItem != null && this.mobjForeCB.Items.Contains(this.mobjListView.SelectedItem.BackColor))
            {
                this.mobjBackCB.SelectedItem = this.mobjListView.SelectedItem.BackColor;
            }
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ListView/Programming/ItemFormatingValidatingPage.vb
Imports System.Drawing
Namespace CompanionKit.Controls.ListView.Programming

	Public Class ItemFormatingValidatingPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

            ' Set DataSource as array of Colors.
			Dim knownColors As KnownColor() = DirectCast(Global.System.Enum.GetValues(GetType(KnownColor)), KnownColor())
			Dim colors As Color() = New Drawing.Color(knownColors.Length - 1) {}
			Dim i As Integer
			For i = 0 To knownColors.Length - 1
				colors(i) = Color.FromKnownColor(knownColors(i))
			Next i
            Me.mobjForeCB.DataSource = colors
            Me.mobjForeCB.ColorMember = "Name"
            Me.mobjForeCB.DisplayMember = "Name"

            Me.mobjBackCB.DataSource = colors.Clone
            Me.mobjBackCB.ColorMember = "Name"
            Me.mobjBackCB.DisplayMember = "Name"

            'Show Grid Lines
            Me.mobjListView.GridLines = True

            ' Fill up the ListView
            Me.mobjListView.Items.Add(New ListViewItem(New String() {"User1", "8-800-1234556"}))
            Me.mobjListView.Items.Add(New ListViewItem(New String() {"User2", "8-800-9513546"}))
            Me.mobjListView.Items.Add(New ListViewItem(New String() {"User3", "8-800-8524563"}))
            Me.mobjListView.Items.Add(New ListViewItem(New String() {"User4", "8-800-9874613"}))
            Me.mobjListView.Items.Add(New ListViewItem(New String() {"User5", "8-800-1234556"}))
            Me.mobjListView.Items.Add(New ListViewItem(New String() {"User6", "8-800-9513546"}))
            Me.mobjListView.Items.Add(New ListViewItem(New String() {"User7", "8-800-8524563"}))
            Me.mobjListView.Items.Add(New ListViewItem(New String() {"User8", "8-800-9874613"}))
            Me.mobjListView.Items.Add(New ListViewItem(New String() {"User9", "8-800-1234556"}))
            Me.mobjListView.Items.Add(New ListViewItem(New String() {"User10", "8-800-9513546"}))

            'Set default background and foreground colors
            For Each objItem As ListViewItem In Me.mobjListView.Items
                objItem.ForeColor = Color.Black
                objItem.BackColor = Color.White
            Next

		End Sub

        ''' <summary>
        ''' Handles the ItemFormatting event of the mobjListView control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="global.Gizmox.WebGUI.Forms.ListViewItemFormattingEventArgs"/> instance containing the event data.</param>
        Private Sub mobjListView_ItemFormatting(ByVal sender As System.Object, ByVal e As Global.Gizmox.WebGUI.Forms.ListViewItemFormattingEventArgs) Handles mobjListView.ItemFormatting

            'Define currently formatted item in ListView
            Dim formattedItem As ListViewItem = Me.mobjListView.SelectedItem
            If formattedItem IsNot Nothing Then
                'Change Result Text
                Me.mobjResultLabel.Text = String.Format("Item #{0} has foreground {1} and background {2} colors.", formattedItem.Index, formattedItem.ForeColor.Name, formattedItem.BackColor.Name)
            End If

        End Sub

        ''' <summary>
        ''' Handles the SelectedIndexChanged event of the mobjForeCB control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub mobjForeCB_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjForeCB.SelectedIndexChanged

          'Change fore color of selected item
            If Me.mobjListView.SelectedItem IsNot Nothing Then
                Me.mobjListView.SelectedItem.ForeColor = DirectCast(Me.mobjForeCB.SelectedItem, Color)
            End If

        End Sub

        ''' <summary>
        ''' Handles the SelectedIndexChanged event of the ComboBox2 control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub mobjBackCB_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjBackCB.SelectedIndexChanged

            'Change back color of selected item
            If Me.mobjListView.SelectedItem IsNot Nothing Then
                Me.mobjListView.SelectedItem.BackColor = DirectCast(Me.mobjBackCB.SelectedItem, Color)
            End If

        End Sub

        ''' <summary>
        ''' Handles the SelectedIndexChanged event of the ListView1 control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub mobjListView_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjListView.SelectedIndexChanged
            'Change selected items in ComboBoxes
            If Me.mobjListView.SelectedItem IsNot Nothing AndAlso Me.mobjForeCB.Items.Contains(Me.mobjListView.SelectedItem.ForeColor) Then
                Me.mobjForeCB.SelectedItem = Me.mobjListView.SelectedItem.ForeColor
            End If
            If Me.mobjListView.SelectedItem IsNot Nothing AndAlso Me.mobjForeCB.Items.Contains(Me.mobjListView.SelectedItem.BackColor) Then
                Me.mobjBackCB.SelectedItem = Me.mobjListView.SelectedItem.BackColor
            End If
        End Sub
	End Class

End Namespace